runtime/ops/command.rs

```diff
@@ -0,0 +1,302 @@
+use super::io::ChildStderrResource;
+use super::io::ChildStdinResource;
+use super::io::ChildStdoutResource;
+use crate::permissions::Permissions;
+use deno_core::error::type_error;
+use deno_core::error::AnyError;
+use deno_core::op_sync;
+use deno_core::AsyncMutFuture;
+use deno_core::AsyncRefCell;
+use deno_core::Extension;
+use deno_core::OpState;
+use deno_core::RcRef;
+use deno_core::Resource;
+use deno_core::ResourceId;
+use deno_core::{op_async, ZeroCopyBuf};
+use serde::Deserialize;
+use serde::Serialize;
+use std::borrow::Cow;
+use std::cell::RefCell;
+use std::rc::Rc;
+use tokio::process::Command;
+
+#[cfg(unix)]
+use std::os::unix::prelude::ExitStatusExt;
+use std::process::ExitStatus;
+
+pub fn init() -> Extension {
+  Extension::builder()
+    .ops(vec![
+      ("op_command_spawn", op_sync(op_command_spawn)),
+      ("op_command_status", op_async(op_command_status)),
+      ("op_command_output", op_async(op_command_output)),
+      ("op_command_child_wait", op_async(op_command_child_wait)),
+      ("op_command_child_output", op_async(op_command_child_output)),
+      ("op_command_child_status", op_sync(op_command_child_status)),
+    ])
+    .build()
+}
+
+struct ChildResource(AsyncRefCell<tokio::process::Child>);
+
+impl Resource for ChildResource {
+  fn name(&self) -> Cow<str> {
+    "child".into()
+  }
+}
+
+impl ChildResource {
+  fn borrow_mut(self: Rc<Self>) -> AsyncMutFuture<tokio::process::Child> {
+    RcRef::map(self, |r| &r.0).borrow_mut()
+  }
+}
+
+fn subprocess_stdio_map(s: &str) -> Result<std::process::Stdio, AnyError> {
+  match s {
+    "inherit" => Ok(std::process::Stdio::inherit()),
+    "piped" => Ok(std::process::Stdio::piped()),
+    "null" => Ok(std::process::Stdio::null()),
+    _ => Err(type_error("Invalid resource for stdio")),
+  }
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandArgs {
+  cmd: String,
+  args: Vec<String>,
+  cwd: Option<String>,
+  clear_env: bool,
+  env: Vec<(String, String)>,
+  stdin: Option<String>,
+  stdout: Option<String>,
+  stderr: Option<String>,
+}
+
+fn create_command(
+  state: &mut OpState,
+  command_args: CommandArgs,
+) -> Result<Command, AnyError> {
+  super::check_unstable(state, "Deno.Command");
+  state
+    .borrow_mut::<Permissions>()
+    .run
+    .check(&command_args.cmd)?;
+
+  let mut command = Command::new(&command_args.cmd);
+  command.args(&command_args.args);
+
+  if let Some(cwd) = command_args.cwd {
+    command.current_dir(cwd);
+  }
+
+  if command_args.clear_env {
+    command.env_clear();
+  }
+  command.envs(command_args.env);
+
+  if let Some(stdin) = &command_args.stdin {
+    command.stdin(subprocess_stdio_map(stdin)?);
+  }
+  if let Some(stdout) = &command_args.stdout {
+    command.stdout(subprocess_stdio_map(stdout)?);
+  }
+  if let Some(stderr) = &command_args.stderr {
+    command.stderr(subprocess_stdio_map(stderr)?);
+  }
+
+  // We want to kill child when it's closed
+  command.kill_on_drop(true);
+
```

Comment:
Add a TODO here regarding detached processes and issue about orphaned child processes when Deno exits with error or `Deno.exit()` is called.